### PR TITLE
Adding content item alternates based on the alias and slug

### LINKED ISSUE #2866: Alternates per content item
My scenario is that I have create pages using Page contenttype. I want these pages to use different template to rendered. e.g. there is a page named userlogin and another named userregister. I want my custom login template to render userlogin page and my custom register template to render userregister page.

I remember in O1, there was an alternates that looks like Content-Item-[ContentId]. But this has to depend on the content item's id. And another approach is using Alias in placementinfo file to specify a alternates. But this seems not be implemented according this document https://orchardcore.readthedocs.io/en/latest/OrchardCore/OrchardCore.DisplayManagement/README/
in OC.

Is there another way to do that?

### PR DESCRIPTION
**Pull request by barthamark**
Fixes #2866

Adds alternates based on the alias or the slug.

### CODE REVIEW

**Review by sebastienros — APPROVED** [review]

File: src/docs/reference/modules/Templates/README.md
Comment (Piedone) [inline src/docs/reference/modules/Templates/README.md:1]: Please add a note about this to https://github.com/OrchardCMS/OrchardCore/blob/main/src/docs/releases/2.1.0.md. See https://docs.orchardcore.net/en/latest/guides/contributing/contributing-documentation/.

**Review by Piedone — CHANGES_REQUESTED** [review]
I didn't review the PR after Sebastien, but am only adding the release notes note.

File: src/docs/reference/modules/Templates/README.md
Comment (barthamark) [reply src/docs/reference/modules/Templates/README.md:1]: I added a note but I'm not sure if it's in the proper section, someone please confirm.

**Review by Piedone — APPROVED** [review]

File: src/docs/releases/2.1.0.md [hunk 1]
```diff
@@ -80,3 +80,7 @@ A new feature was added to allow you to send SMS messages using Azure Communicat
 ### External Authentication Feature
 
 The new **External Authentication** feature contains common functionality used by multiple external (i.e. non-local) authentication providers (like Microsoft or Google login). See [above](#external-authentication-logic-has-been-separated-from-the-users-feature) for details.
+
+### Content Item Shape Alternates Based on Alias and Slug
```
Comment (MikeAlhayek) [inline src/docs/releases/2.1.0.md:84]: This should have been added before line 74 since it it not a new feature

**Pull request merged**